xdsl/backend/riscv/prologue_epilogue_insertion.py

```diff
@@ -42,6 +42,9 @@ def _process_function(self, func: riscv_func.FuncOp) -> None:
             if res.type in Registers.S or res.type in Registers.FS
         )
 
+        if len(used_callee_preserved_registers) == 0:
```

Comment:
```suggestion
        if not used_callee_preserved_registers:
```
This is the idiomatic way to check if a sequence is empty in Python, it avoids a potentially costly length computation.

Reply:
Sure, I'm not yet completely familiar with those 👍